core/flow/traffic_shaping_test.go

```diff
@@ -0,0 +1,142 @@
+package flow
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/core/stat"
+)
+
+func TestTrafficShapingController_Rule(t *testing.T) {
+	type fields struct {
+		flowCalculator TrafficShapingCalculator
+		flowChecker    TrafficShapingChecker
+		rule           *FlowRule
+	}
+
+	rule := []byte(`{"resource":"Sentinel","limitApp":"","grade":1,"count":0,"strategy":0,"controlBehavior":0,"warmUpPeriodSec":10,"maxQueueingTimeMs":1000,"clusterMode":true,"clusterConfig":{"thresholdType":1}}`)
+	flowRule := &FlowRule{}
+	_ = json.Unmarshal(rule, &flowRule)
+
+	tests := []struct {
+		name   string
+		fields fields
+		want   *FlowRule
+	}{
+		{"test1", fields{nil, nil, flowRule}, flowRule},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			traffic := &TrafficShapingController{
+				flowCalculator: tt.fields.flowCalculator,
+				flowChecker:    tt.fields.flowChecker,
+				rule:           tt.fields.rule,
+			}
+			if got := traffic.Rule(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("TrafficShapingController.Rule() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestTrafficShapingController_FlowChecker(t *testing.T) {
+	type fields struct {
+		flowCalculator TrafficShapingCalculator
+		flowChecker    TrafficShapingChecker
+		rule           *FlowRule
+	}
+
+	flowChecker := NewThrottlingChecker(1000)
+
+	tests := []struct {
+		name   string
+		fields fields
+		want   TrafficShapingChecker
+	}{
+		{"test1", fields{nil, flowChecker, nil}, flowChecker},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			traffic := &TrafficShapingController{
+				flowCalculator: tt.fields.flowCalculator,
+				flowChecker:    tt.fields.flowChecker,
+				rule:           tt.fields.rule,
+			}
+			if got := traffic.FlowChecker(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("TrafficShapingController.FlowChecker() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestTrafficShapingController_PerformChecking(t *testing.T) {
+	type fields struct {
+		flowCalculator TrafficShapingCalculator
+		flowChecker    TrafficShapingChecker
+		rule           *FlowRule
+	}
+	type args struct {
+		node         base.StatNode
+		acquireCount uint32
+		flag         int32
+	}
+
+	flowCalculator := NewDefaultTrafficShapingCalculator(99)
```

Comment:
It's better to mock interface TrafficShapingCalculator and TrafficShapingChecker.
The usage of mock object, you can refer to "github.com/stretchr/testify/mock".

The principle of unit test:
Unit test should ensure that the down layer logic is isolated.